migrations/20171024163252-host-cleanup.js

```diff
@@ -0,0 +1,318 @@
+            } 
+
+            const tSplit = splitTransactionGroup(transactions);
+
+            // I'm sure there is a way to do this in one query... 
+            return sequelize.query(`
+              UPDATE "Transactions"
+                SET "deletedAt" = :date
+              WHERE id = :tId
+              `, {
+                replacements: {
+                  date: new Date(),
+                  tId: tSplit.withoutFromCollective.id
+                }
+              })
+              .then(() => sequelize.query(`
+                UPDATE "Transactions"
+                  SET "CollectiveId" = :orgCollectiveId
+                WHERE id= :tId
+                `, {
+                  replacements: {
+                    orgCollectiveId: orgCollective.id,
+                    tId: tSplit.debit.id
+                  }
+                }))
+                .then(() => sequelize.query(`
+                  UPDATE "Transactions"
+                    SET "FromCollectiveId" = :orgCollectiveId
+                  WHERE id= :tId
+                  `, {
+                    replacements: {
+                      orgCollectiveId: orgCollective.id,
+                      tId: tSplit.credit.id
+                    }
+                  }))
+          })
+      })
+  }
+
+  return sequelize.query(`
+    SELECT * from "Collectives"
+    WHERE "isSupercollective" is true
+    `, { type: sequelize.QueryTypes.SELECT})
+  .then(superCollectives => {
+    console.log('>>> supercollectives found: ', superCollectives.length);
+    return superCollectives;
+  })
+  // only return those that are using themselves as host
+  .filter(superCollective => superCollective.HostCollectiveId === superCollective.id)
+  .then(superCollectivesSubset => {
+    console.log('>>> supercollectives need to be fixed: ', superCollectivesSubset.length);
+    console.log('>>> supercollectives slugs: ', superCollectivesSubset.map(c => c.slug))
+    return superCollectivesSubset;
+  })
+  .each(superCollective => {
+    console.log('>>> Processing', superCollective.slug);
+    let orgCollective;
+    return createOrg(superCollective)
+      .then(org => {
+        orgCollective = org;
+        return addAdmins(superCollective, orgCollective)
+      })
+      .then(() => makeOrgHost(superCollective, orgCollective))
+      .then(() => moveStripeAccount(superCollective, orgCollective))
+      .then(() => movePaypalAccount(superCollective, orgCollective))
+      .then(() => fixTransactions(superCollective, orgCollective))
+    })
+}
+
+
+const fixHostCollectiveIds = (sequelize) => {
+
+  // a host shouldn't be listed as it's own HostCollectiveId 
+  return sequelize.query(`
+    WITH hosts AS 
+      (SELECT DISTINCT("HostCollectiveId") from "Collectives")
+
+    UPDATE "Collectives" 
+      SET "HostCollectiveId" = null
+    WHERE 
+      id IN (SELECT * FROM hosts);
+    `)
+  
+  // Remove HostCollectiveId from Events, only need ParentCollectiveId
```

Comment:
@xdamman is this going to break something? I see that lots of events have "HostCollectiveId set.

Reply:
It's fine. There is no need. It doesn't harm either way.